packages/server/src/AccountsServer.js

```diff
@@ -176,13 +176,30 @@ export class AccountsServer {
       throw new AccountsError('Email already exists', { email: user.email });
     }
 
-    // TODO Accounts.onCreateUser
-    const userId: string = await this.db.createUser({
+    const { validateNewUser, onUserCreated } = this.options();
+
+    const proposedUserObject = {
       username: user.username,
       email: user.email && user.email.toLowerCase(),
       password: user.password,
       profile: user.profile,
-    });
+    };
+
+    let userIsValid = true;
+    if (isFunction(validateNewUser)) {
+      userIsValid = await validateNewUser(proposedUserObject);
+    }
+
+    if (!userIsValid) {
+      throw new AccountsError('User did not pass validation');
```

Comment:
I would much rather the validateNewUser to return a rejected promise. the code here we could do `await` and it will basically throw. Only thing to think about is whether or not the error should be wrapped as an `AccountsError`